contributors/design-proposals/live-and-inplace-vertical-scaling.md

```diff
@@ -0,0 +1,325 @@
+It is desirable to submit a change in policy, confirm its acceptance (by re-reading the changed pod spec) and then submit a change to a resource size impacted by that policy. 
+
+# Design and Implementation
+
+This section highlights some of implementation details by describing API changes and the workflow of vertical scaling on a pod.
+
+## Changes on API and key components
+
+This section describes briefly API changes for pod-level vertical scaling and the related changes on the 'API server', the 'Scheduler', and the 'Kubelet'.
+
+* **ResizeRequest**
+
+A new data structure, ResizeRequest, is added to v1.PodSpec:
+
+```go
+const (
+    ResizeRequested ResizeStatus = "Requested"
+    ResizeAccepted ResizeStatus = “Accepted”
+    ResizeRejected ResizeStatus = “Rejected”
+    ResizeNone      ResizeStatus = "None"
+)
+
+type ResizeRequest struct {
+    RequestStatus ResizeStatus
+    NewResources  []ResourceRequirements // indexed by containers’ index
+}
+
+Type PodSpec {
+    …
+    ResizeRequest ResizeRequest
+    ...
+}
+```
+
+ResizeRequest has two variables, RequestStatus and NewResources.
+RequestStatus represents the status of a resource resizing request.
+ResizeRequested indicates resource resizing for a pod is requested.
+ResizedAccepted and ResizedRejected means that the requested resource resizing is accepted and rejected, respectively, by the Scheduler.
+The NewResources is an array indexed by a container’s index and its each entry holds new resource requirements of a container that needs to resize.
+
+Given a new PodSpec with new resource requirements from a client, first the 'API server' validates it.
+If it is valid, the 'API server' sets the RequestStatus to ResizedRequested and copies the new resource requirements of each container into the NewResources.
+Also, the 'API server' restores the resource requirements of each container of the PodSpec to the original and writes the revised PodSpec to ETCD to communicate with the Scheduler.
+This is because at this moment the PodSpec on ETCD shouldn’t be updated with new resource requirements.
+
+For a pod with ResizeRequested, the 'Scheduler' checks if the node on which the pod currently runs has enough resources to resize the pod.
+The Scheduler notify the 'API server' of the result via 'Resizing' API operation, which will be describe below.
+
+* **Resizing**
+
+A new API, Resizing, for the 'scheduler' is introduced:
+
+```go
+// Resizing resizes the resources allocated to a pod
+type Resizing struct {
+    metav1.TypeMeta
+    metav1.ObjectMeta
+    Request ResizeRequest
+}
+```
+
+Resizing has the metadata of a pod to resize and a value of ResizeRequest that holds the status of a resizing request, which indicates whether the resizing is feasible or not, and new resource requirements of the pod.
+
+Once the 'Scheduler' determine whether a resource resizing on a pod is feasible, or not, it notifies to the API server via this Resizing API.
```

Comment:
Could we add a policy that scheduler just waits for another retry to determine instead of rejecting it immediately? And user could specify this policy in ResizeRequest.